open_spiel/python/games/dynamic_routing/mean_field_routing_game.py

```diff
@@ -0,0 +1,547 @@
+                        self.assert_valid_action(action, self._vehicle_location)
+                    self._vehicle_location =\
+                        self.get_game().network.get_road_section_from_action_id(
+                            action)
+                    # Has the vehicle just reached its destination?
+                    if (self._vehicle_location ==
+                            self._vehicle_destination):
+                        self._vehicle_final_travel_time =\
+                          self._current_time_step
+                        self._vehicle_at_destination = True
+                        self._vehicle_without_legal_action = True
+                    # Will the vehicle have a legal action for next time step?
+                    if self.get_game().network.is_location_at_sink_node(
+                            self._vehicle_location):
+                        self._vehicle_without_legal_action = True
+            self._current_time_step += 1
+        # Is the game finished?
+        if self._current_time_step >= self.get_game().max_game_length():
+            self._is_terminal = True
+            if not self._vehicle_at_destination:
+                self._vehicle_final_travel_time =\
+                  - self.get_game().min_utility()
+
+    def _action_to_string(self, player, action) -> str:
+        """Action -> string."""
+        if player == pyspiel.PlayerId.CHANCE:
+            if self._is_chance_init:
+                return f"Vehicle is assigned to population {action}."
+            return f"Change node; the vehicle movement is {bool(action)}."
+        if self.get_game().perform_sanity_checks:
+            assert player == pyspiel.PlayerId.DEFAULT_PLAYER_ID
+        if action == NO_POSSIBLE_ACTION:
+            return f"Vehicle {player} reach a sink node or its destination."
+        if self.get_game().perform_sanity_checks:
+            self.assert_valid_action(action)
+        return (f"Vehicle {player} would like to move to " +
+                str(self.get_game().network.get_road_section_from_action_id(
+                    action)) + ".")
+
+    def is_terminal(self) -> bool:
+        """Returns True if the game is over."""
+        return self._is_terminal
+
+    def returns(self) -> List[float]:
+        """Total reward for each player over the course of the game so far."""
+        if not self._is_terminal:
+            return [0]
+        return [- self._vehicle_final_travel_time]
+
+    def get_location_as_int(self) -> int:
+        """Get the vehicle location."""
+        if not hasattr(self, '_vehicle_location'):
+            return -1
+        start_node, end_node = _road_section_to_nodes(self._vehicle_location)
+        return self.get_game().network.get_action_id_from_movement(
+            start_node, end_node)
+
+    def __str__(self) -> str:
+        """String for debug purposes. No particular semantics are required."""
+        if hasattr(self, '_vehicle_location'):
```

Comment:
Set it to None in __init__(), and check that that here. It will be more robust (e.g. to spelling mistakes).